Storage/api.h

```diff
@@ -480,11 +553,119 @@ class RESTfulStorage {
   };
 
   void RegisterRoute(const std::string& field_name, const RESTfulRoute& route) {
-    const auto path = route_prefix_ + '/' + route.resource_prefix + '/' + field_name + route.resource_suffix;
-    handler_routes_.emplace_back(path, route.resource_args_mask);
-    handlers_scope_ += HTTP(port_).Register(path, route.resource_args_mask, route.handler);
+    const auto path = data_->route_prefix_ + '/' + route.resource_prefix +
+                      (field_name.empty() ? "" : '/' + field_name) + route.resource_suffix;
+    data_->handler_routes_.emplace_back(path, route.resource_args_mask);
+    data_->handlers_scope_ += HTTP(data_->port_).Register(path, route.resource_args_mask, route.handler);
   }
 
+  template <typename T>
+  static std::shared_ptr<CurrentSuper> ParseCQRSRequest(const Request& request) {
+    // TODO(dkorolev): Proper exception types here!
+    auto object = std::make_shared<T>();
+    if (reflection::FieldCounter<T>::value > 0) {
+      if (!request.body.empty()) {
+        ParseJSON(request.body, *object);
+      } else {
+        request.url.query.FillObject(*object);
```

Comment:
So, your plan allows to pass the same command payload as a query string or as body, to the discretion of the caller. This may result in sensitive data (that is I assume mostly passed via body) to get into query string.